meta-mender-core/recipes-bsp/grub-mender-grubenv/grub-mender-grubenv_1.1.0.bb

```diff
@@ -60,13 +79,37 @@ mender_grub_storage_device=$mender_grub_storage_device
 kernel_imagetype=${KERNEL_IMAGETYPE}
 EOF
 
+    if ${@bb.utils.contains('DISTRO_FEATURES', 'mender-uuid', 'true', 'false', d)}; then
+        if ${@bb.utils.contains('DISTRO_FEATURES', 'mender-image-gpt', 'true', 'false', d)} || ${@bb.utils.contains('DISTRO_FEATURES', 'mender-image-uefi', 'true', 'false', d)}; then
+            mender_rootfsa_uuid=${MENDER_ROOTFS_PART_A_UUID}
+            mender_rootfsb_uuid=${MENDER_ROOTFS_PART_B_UUID}
+        else
+            mender_rootfsa_uuid="${MENDER_DISK_IDENTIFIER}-${@get_part_nummber_padded_from_device(MENDER_ROOTFS_PART_A)}"
+            mender_rootfsb_uuid="${MENDER_DISK_IDENTIFIER}-${@get_part_nummber_padded_from_device(MENDER_ROOTFS_PART_B)}"
+        fi
+
+        cat >> ${B}/mender_grubenv_defines <<EOF
+mender_rootfsa_uuid=$mender_rootfsa_uuid
+mender_rootfsb_uuid=$mender_rootfsb_uuid
+EOF
+    fi
+
     if [ -n "${KERNEL_DEVICETREE}" ]; then
         MENDER_DTB_NAME=$(mender_get_clean_kernel_devicetree)
         cat >> ${B}/mender_grubenv_defines <<EOF
 kernel_devicetree=$MENDER_DTB_NAME
 EOF
     fi
 }
+do_configure[vardeps] = " \
+    MENDER_ROOTFS_PART_A \
+    MENDER_ROOTFS_PART_B \
+    MENDER_STORAGE_DEVICE_BASE \
+    MENDER_ROOTFS_PART_A_UUID \
+    MENDER_ROOTFS_PART_B_UUID \
+    MENDER_DISK_IDENTIFIER \
+    KERNEL_DEVICETREE \
+"
```

Comment:
This should not be needed when you are making direct references to variables using the `${VAR}` notation. It will discover the references automatically.